### ISSUE TITLE
Batch norm backward pass isn't right

### ISSUE THREAD

**Issue by geohot** [geohot/tinygrad#141]
Fix!

**Comment by marcelbischoff** [comment]
So batchnorm is different for training and inference. We don't have a training flag, do we?

**Comment by geohot** [comment]
It's not a global flag, and I don't want it to be. The BatchNorm class should support it, and we should have good helpers to find instances of a class type to change it.

**Comment by geohot** [comment]
I see progress in #147

**Comment by marcelbischoff** [comment]
Several issues:
- torch has `training` and `track_running_stats` and `momentum`
- easiest case is training where one just compute the batch mean and variance
- but computing the variance is very unstable and the naive implementation gives different result from numpy so a test for this simple case does not pass any reasonable tolerance (maybe there is another bug)

**Comment by geohot** [comment]
I think the first PR to do here is to add axis support to sum and mean. Then the batchnorm code will be readable and we can go from there.

**Comment by marcelbischoff** [comment]
We essentially have that, it's just broadcasting with `forward` <-> `backward`

**Comment by geohot** [comment]
No, I mean add the arg to the ops. I agree we mostly have it, but it should be in the op.

**Comment by marcelbischoff** [comment]
Yes I understand, I am wondering if one can save some lines because it is the same operation (differently called)

**Comment by marcelbischoff** [comment]
backward seems to work now!
